.konflux/catalog/.gitignore

```diff
@@ -0,0 +1 @@
+numaresources-operator/catalog.yaml
```

Comment:
I'm unsure if we should keep `.gitkeep` if we are explicit about the files to ignore in the `numaresources-operator` directory